specification/common-types/resource-management/v4/customermanagedkeys.json

```diff
@@ -0,0 +1,70 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "4.0",
+    "title": "Common types"
+  },
+  "paths": {},
+  "definitions": {
+    "encryption": {
+      "type": "object",
+      "description": "(Optional) Discouraged to include in resource definition. Only needed where it is possible to disable platform (AKA infrastructure) encryption. Azure SQL TDE is an example of this. Values are enabled and disabled.",
+      "properties": {
+        "infrastructureEncryption": {
+          "type": "string",
+          "enum": [
+            "enabled",
+            "disabled"
+          ],
+          "x-ms-enum": {
+            "name": "InfrastructureEncryption",
+            "modelAsString": true
+          },
+          "description": "Values are enabled and disabled."
+        },
+        "customerManagedKeyEncryption": {
+          "description": "All Customer-managed key encryption properties for the resource.",
+          "$ref": "#/definitions/customerManagedKeyEncryption"
+        }
+      }
+    },
+    "customerManagedKeyEncryption": {
+      "type": "object",
+      "description": "All Customer-managed key encryption properties for the resource.",
+      "properties": {
+        "keyEncryptionKeyIdentity": {
+          "type": "object",
+          "description": "All identity configuration for Customer-managed key settings defining which identity should be used to auth to Key Vault.",
+          "properties": {
+            "identityType": {
+              "type": "string",
+              "enum": [
+                "systemAssignedIdentity",
+                "userAssignedIdentity",
+                "none"
```

Comment:
>"none" [](http://example.com/codeflow?start=16&length=6)

Is none expected? what happens if a user sends "none", what happens if a user doesn't pass this property? can a user update this value after it has been created? suggest removing "none" and treat "null" as the default value to avoid confusions, since there is no identityType equals to none.

Reply:
Note we actually have 3 identity types supported, SystemAssigned, UserAssigned and DelegatedResource which should be only used internally by other RPs, to allow CMK with HOBO-V2 (as defined [here](https://armwiki.azurewebsites.net/authorization/managed_identities/hosted_on_behalf_of_v2/Overview.html) and [here](https://armwiki.azurewebsites.net/authorization/managed_identities/hosted_on_behalf_of_v2/BackingResourceProviderOnboarding.html)).
We are trying to avoid explicitly exposing the third option in public, so we are planning to treat the type 'none' as supported with delegatedResource only (by passing delegatedIdentltyClientId). We named it 'none' because this is the third IdentityType value defined by MSI in the general [Identity schema](https://armwiki.azurewebsites.net/authorization/managed_identities/MSIOnboardingSteps.html).

If you have a better suggestion for how to support delegatedResource type internally only, we would love to hear that

Reply:
@jackrichins what do you think about that?
How should be represent the usage of delegated identity in CMK?
Should we use identityType=null, 'none', or maybe 'delegatedIdentity' type?

Reply:
I asked Storage how they handle delegatedIdentity and they responded with the following:
We have an AFEC – [Microsoft.Storage](http://microsoft.storage/)\AllowSharedIdentities that first party services should onboard onto in order to use the property. Only subscriptions in the Microsoft/AME tenant are whitelisted to register the AFEC. Also, the property is not exposed to any external customers and is not part of the external REST API specifications.
To learn more, Manu Suresh Rao <manusrao@microsoft.com> is the engineer who owns it for Storage.

Reply:
It's just not enough information. When the type is systemAssignedIdentity, the RP can't possibly know if it's the systemAssignedIdentity of this RP, or of another RP that was delegating to this RP.
For userAssignedIdentity, you could say that the RP can take the userAssignedIdentityResourceId and try to match in it's own userAssignedIdentities or with one of the delegated identities it accepted.

Reply:
@jackrichins note that that I've changed the identityType value from 'none' to 'delegatedResourceIdentity' to be more clear . We also found this is the value used in other team supporting CMK with delegated identities